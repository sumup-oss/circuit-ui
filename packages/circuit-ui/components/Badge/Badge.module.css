.base {
  display: inline-block;
  padding: 2px var(--cui-spacings-byte);
  font-size: var(--cui-body-s-font-size);
  font-weight: var(--cui-font-weight-bold);
  line-height: var(--cui-body-s-line-height);
  text-align: center;
  letter-spacing: 0.25px;
  border-radius: var(--cui-border-radius-pill);
}

.circle {
  display: flex;
  align-items: center;
  justify-content: center;
  width: var(--badge-width);
  height: 24px;
  padding: 2px 4px;
}

/* Variants */

.success {
  color: var(--cui-fg-on-strong);
  background-color: var(--cui-bg-success-strong);
}

.warning {
  color: var(--cui-fg-on-strong);
  background-color: var(--cui-bg-warning-strong);
}

.danger {
  color: var(--cui-fg-on-strong);
  background-color: var(--cui-bg-danger-strong);
}

.neutral {
  color: var(--cui-fg-normal);
  background-color: var(--cui-bg-highlight);
}

.promo {
  color: var(--cui-fg-on-strong);
  background-color: var(--cui-bg-promo-strong);
}
