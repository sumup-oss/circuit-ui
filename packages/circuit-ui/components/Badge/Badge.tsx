/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Ref, HTMLAttributes } from 'react';
import { css } from '@emotion/react';

import styled, { StyleProps } from '../../styles/styled';
import { typography } from '../../styles/style-mixins';
import { withDeprecation } from '../../util/logger';

export interface BadgeProps extends HTMLAttributes<HTMLDivElement> {
  /**
   * Choose the style variant. Default: 'neutral'.
   */
  variant?:
    | 'neutral'
    | 'success'
    | 'warning'
    | 'danger'
    | 'promo'
    /**
     * @deprecated
     */
    | 'confirm'
    /**
     * @deprecated
     */
    | 'notify'
    /**
     * @deprecated
     */
    | 'alert';
  /**
   * Use the circular badge to indicate a count of items related to an element.
   */
  circle?: boolean;
  /**
   * The ref to the HTML DOM element.
   */
  ref?: Ref<HTMLDivElement>;
}

const baseStyles = ({ theme }: StyleProps) => css`
  border-radius: ${theme.borderRadius.pill};
  display: inline-block;
  padding: 2px ${theme.spacings.byte};
  font-weight: ${theme.fontWeight.bold};
  text-align: center;
  letter-spacing: 0.25px;
`;

const variantStyles = ({ variant = 'neutral' }: BadgeProps) => {
  switch (variant) {
    case 'success':
    case 'confirm': {
      return css`
        background-color: var(--cui-bg-success-strong);
        color: var(--cui-fg-on-strong);
      `;
    }
    case 'warning':
    case 'notify': {
      return css`
        background-color: var(--cui-bg-warning-strong);
        color: var(--cui-fg-on-strong);
      `;
    }
    case 'danger':
    case 'alert': {
      return css`
        background-color: var(--cui-bg-danger-strong);
        color: var(--cui-fg-on-strong);
      `;
    }
    case 'neutral': {
      return css`
        background-color: var(--cui-bg-highlight);
        color: var(--cui-fg-normal);
      `;
    }
    case 'promo': {
      return css`
        background-color: var(--cui-bg-promo-strong);
        color: var(--cui-fg-on-strong);
      `;
    }
    default: {
      return null;
    }
  }
};

const isDynamicWidth = (children: BadgeProps['children']) => {
  if (typeof children === 'string') {
    return children.length > 2;
  }
  return false;
};

const circleStyles = ({ circle = false, children }: BadgeProps) =>
  circle &&
  css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 2px 4px;
    height: 24px;
    width: ${isDynamicWidth(children) ? 'auto' : '24px'};
  `;

/**
 * A badge communicates the status of an element or the count of items
 * related to an element.
 */
const StyledBadge = styled('div')<BadgeProps>(
  typography('two'),
  baseStyles,
  variantStyles,
  circleStyles,
);

StyledBadge.displayName = 'Badge';

export const Badge =
  process.env.NODE_ENV === 'production'
    ? StyledBadge
    : withDeprecation(StyledBadge, ({ variant }) => {
        const deprecatedMap: Record<string, string> = {
          confirm: 'success',
          notify: 'warning',
          alert: 'danger',
        };

        if (variant && deprecatedMap[variant]) {
          return `The "${variant}" variant has been deprecated. Use "${deprecatedMap[variant]}" instead.`;
        }

        return null;
      });
