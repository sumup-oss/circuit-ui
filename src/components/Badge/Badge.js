/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css } from '@emotion/core';
import { size } from 'polished';

import { subHeadingKilo } from '../../styles/style-helpers';
import { colorNames } from '../../styles/constants';

const COLOR_MAP = {
  [colorNames.SUCCESS]: {
    default: 'g500',
    hover: 'g700',
    active: 'g900'
  },
  [colorNames.WARNING]: {
    default: 'y500',
    hover: 'y700',
    active: 'y900'
  },
  [colorNames.DANGER]: {
    default: 'r500',
    hover: 'r700',
    active: 'r900'
  },
  [colorNames.PRIMARY]: {
    default: 'b500',
    hover: 'b700',
    active: 'b900'
  },
  [colorNames.NEUTRAL]: {
    default: 'n500',
    hover: 'n700',
    active: 'n900'
  }
};

const colorStyles = ({ theme, color, onClick }) => {
  const currentColor = COLOR_MAP[color];

  if (!currentColor) {
    return null;
  }

  return css`
    label: ${`badge--${color}`};
    background-color: ${theme.colors[currentColor.default]};
    ${onClick &&
      `
      &:hover {
        background-color: ${theme.colors[currentColor.hover]};
      }

      &:active {
        background-color: ${theme.colors[currentColor.active]};
      }
    `};
  `;
};

const baseStyles = ({ theme, onClick }) => css`
  label: badge;
  border-radius: 100px;
  color: ${theme.colors.white};
  cursor: ${onClick ? 'pointer' : 'default'};
  display: inline-block;
  padding: 0 ${theme.spacings.byte};
  ${subHeadingKilo({ theme })};
  font-weight: ${theme.fontWeight.bold};
  text-transform: uppercase;
  user-select: none;
  text-align: center;
`;

const circleStyles = ({ circle }) =>
  circle &&
  css`
    label: badge--circle;
    display: flex;
    align-items: center;
    justify-content: center;
    ${size(24)};
  `;

const StyledBadge = styled('div')`
  ${baseStyles};
  ${colorStyles};
  ${circleStyles};
`;

/**
 * A badge for displaying update notifications etc.
 */
const Badge = props => <StyledBadge {...props} />;

Badge.NEUTRAL = colorNames.NEUTRAL;
Badge.PRIMARY = colorNames.PRIMARY;
Badge.SUCCESS = colorNames.SUCCESS;
Badge.WARNING = colorNames.WARNING;
Badge.DANGER = colorNames.DANGER;

Badge.propTypes = {
  /**
   * Callback for the click event.
   */
  onClick: PropTypes.func,
  /**
   * Ensures text is centered and the badge looks like a circle.
   */
  circle: PropTypes.bool,
  color: PropTypes.oneOf([
    Badge.NEUTRAL,
    Badge.PRIMARY,
    Badge.SUCCESS,
    Badge.WARNING,
    Badge.DANGER
  ])
};

Badge.defaultProps = {
  circle: false,
  color: Badge.NEUTRAL
};

/**
 * @component
 */
export default Badge;
