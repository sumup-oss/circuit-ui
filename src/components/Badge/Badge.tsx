/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, {
  forwardRef,
  MouseEvent,
  KeyboardEvent,
  Ref,
  HTMLProps
} from 'react';
import { css } from '@emotion/core';
import isPropValid from '@emotion/is-prop-valid';

import styled, { StyleProps } from '../../styles/styled';
import { subHeadingKilo, focusOutline } from '../../styles/style-helpers';

type OnClick = (
  event: MouseEvent<HTMLDivElement> | KeyboardEvent<HTMLDivElement>
) => void;
type RefType = Ref<HTMLDivElement>;

export interface BadgeProps extends HTMLProps<HTMLDivElement> {
  /**
   * Callback for the click event.
   */
  onClick?: OnClick;
  /**
   * Ensures text is centered and the badge looks like a circle.
   */
  circle?: boolean;
  color?: 'neutral' | 'primary' | 'success' | 'warning' | 'danger';
  /**
   * The ref to the html div DOM element
   */
  ref?: RefType;
  as?: string;
}

const COLOR_MAP = {
  success: {
    default: 'g500',
    hover: 'g700',
    active: 'g900'
  },
  warning: {
    default: 'y500',
    hover: 'y700',
    active: 'y900'
  },
  danger: {
    default: 'r500',
    hover: 'r700',
    active: 'r900'
  },
  primary: {
    default: 'b500',
    hover: 'b700',
    active: 'b900'
  },
  neutral: {
    default: 'n500',
    hover: 'n700',
    active: 'n900'
  }
} as const;
const baseStyles = ({ theme }: StyleProps) => css`
  label: badge;
  border-radius: ${theme.borderRadius.pill};
  color: ${theme.colors.white};
  display: inline-block;
  padding: 0 ${theme.spacings.byte};
  ${subHeadingKilo({ theme })};
  font-weight: ${theme.fontWeight.bold};
  text-transform: uppercase;
  user-select: none;
  text-align: center;
`;

const colorStyles = ({ theme, color = 'neutral' }: StyleProps & BadgeProps) => {
  const currentColor = COLOR_MAP[color];
  if (!currentColor) {
    return null;
  }
  return css`
    label: ${`badge--${color}`};
    background-color: ${theme.colors[currentColor.default]};
  `;
};

const circleStyles = ({ circle }: BadgeProps) =>
  circle &&
  css`
    label: badge--circle;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 24px;
    width: 24px;
  `;

const clickableStyles = ({
  theme,
  onClick,
  color = 'neutral'
}: StyleProps & BadgeProps) => {
  const currentColor = COLOR_MAP[color];
  if (!onClick || !currentColor) {
    return null;
  }
  return css`
    label: badge--clickable;
    border: 0;
    outline: 0;
    cursor: pointer;

    &:hover {
      background-color: ${theme.colors[currentColor.hover]};
    }

    &:active {
      background-color: ${theme.colors[currentColor.active]};
    }

    &:focus {
      ${focusOutline({ theme })};
    }
  `;
};

/**
 * A badge for displaying update notifications etc.
 */
const StyledBadge = styled('div', {
  shouldForwardProp: prop => isPropValid(prop) && prop !== 'color'
})<BadgeProps>(baseStyles, colorStyles, circleStyles, clickableStyles);

/* eslint-disable react/display-name */
const Badge = forwardRef((props: BadgeProps, ref: BadgeProps['ref']) => (
  <StyledBadge as={props.onClick ? 'button' : 'div'} ref={ref} {...props} />
));

/**
 * @component
 */
export default Badge;
