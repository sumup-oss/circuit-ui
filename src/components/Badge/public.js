export { default as Badge } from '.';
