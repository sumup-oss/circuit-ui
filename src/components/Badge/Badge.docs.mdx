import { Status, Props, Story } from '../../../.storybook/components';
import Badge from './Badge';

# Badge

<Status.Stable />

Badges are non-actionable elements in the UI, used to indicate the status of an object.

<Story id="components-badge--base" />

<Props of={Badge} />

## Best practices

- Use our colour patterns so that users can learn and identify status at a glance.
- Create a clear label with a short, scannable text.
- Position a badge so that it is obvious what content the status is related to.

## Usage guidelines

- **Do** use badges to indicate the status of data rows in a table.
- **Do** use badges to bring attention to a specific area on the screen (like a
  new feature).
- **Do not** use badges to indicate default statuses (e.g. in the Transaction
  History, a transaction should only include a status if it's anything other
  than default; failed, refunded, paid out, etc.)

## Component variations

### Semantic badges

The badges come in a number of colors based on their semantic meaning.

<Story id="components-badge--colors" />

### Circular badges

Badges that receive the `circle` prop can be used to indicate notifications.

<Story id="components-badge--circular" />
