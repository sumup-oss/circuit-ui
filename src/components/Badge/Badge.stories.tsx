/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';

import { Stack } from '../../../.storybook/components';

import { Badge, BadgeProps } from './Badge';
import docs from './Badge.docs.mdx';

export default {
  title: 'Components/Badge',
  component: Badge,
  parameters: {
    docs: { page: docs },
  },
};

export const Base = (args: BadgeProps) => <Badge {...args} />;

Base.args = {
  children: 'Badge',
};

export const Variants = (args: BadgeProps) => (
  <Stack>
    <Badge {...args} variant="neutral">
      Neutral
    </Badge>
    <Badge {...args} variant="success">
      Success
    </Badge>
    <Badge {...args} variant="warning">
      Warning
    </Badge>
    <Badge {...args} variant="danger">
      Danger
    </Badge>
  </Stack>
);

export const Circular = (args: BadgeProps) => (
  <Stack>
    <Badge {...args} circle>
      1
    </Badge>
    <Badge {...args} circle>
      42
    </Badge>
    <Badge {...args} circle>
      999
    </Badge>
  </Stack>
);
