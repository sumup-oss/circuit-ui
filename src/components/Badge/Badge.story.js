/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { Fragment } from 'react';
import { select, boolean } from '@storybook/addon-knobs/react';
import { values } from 'lodash/fp';

import { colorNames } from '../../styles/constants';

import docs from './Badge.docs.mdx';
import Badge from './Badge';

export default {
  title: 'Components/Badge',
  component: Badge,
  parameters: {
    docs: { page: docs },
    jest: ['Badge']
  }
};

export const base = () => (
  <Badge color={select('Color', values(colorNames))}>Badge</Badge>
);

export const colors = () => (
  <Fragment>
    <Badge color={Badge.NEUTRAL}>Neutral</Badge>
    <Badge color={Badge.PRIMARY}>Primary</Badge>
    <Badge color={Badge.SUCCESS}>Success</Badge>
    <Badge color={Badge.WARNING}>Warning</Badge>
    <Badge color={Badge.DANGER}>Danger</Badge>
  </Fragment>
);

export const circular = () => (
  <Badge color={Badge.PRIMARY} circle={boolean('Circular', true)}>
    42
  </Badge>
);
