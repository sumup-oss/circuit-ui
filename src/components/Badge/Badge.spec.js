/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';

import Badge from '.';

describe('Badge', () => {
  /**
   * Style tests.
   */
  it('should render with default styles', () => {
    const actual = create(<Badge />);
    expect(actual).toMatchSnapshot();
  });

  describe('rendering color variations', () => {
    it('should render with success colors', () => {
      const actual = create(<Badge color={Badge.SUCCESS} />);
      expect(actual).toMatchSnapshot();
    });

    it('should render with warning colors', () => {
      const actual = create(<Badge color={Badge.WARNING} />);
      expect(actual).toMatchSnapshot();
    });

    it('should render with danger colors', () => {
      const actual = create(<Badge color={Badge.DANGER} />);
      expect(actual).toMatchSnapshot();
    });

    it('should render with primary colors', () => {
      const actual = create(<Badge color={Badge.PRIMARY} />);
      expect(actual).toMatchSnapshot();
    });
  });

  it('should have hover/active styles only when the onClick handler is provided', () => {
    const actual = create(<Badge onClick={jest.fn()} />);
    expect(actual).toMatchSnapshot();
  });

  it('should have the correct circle styles', () => {
    const actual = create(<Badge circle />);
    expect(actual).toMatchSnapshot();
  });

  /**
   * Accessibility tests.
   */
  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<Badge />);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });

  /**
   * Logic tests.
   */
  it('should be clickable', () => {
    const onClick = jest.fn();
    const { getByTestId } = render(
      <Badge onClick={onClick} data-testid="badge" />
    );

    act(() => {
      fireEvent.click(getByTestId('badge'));
    });

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
