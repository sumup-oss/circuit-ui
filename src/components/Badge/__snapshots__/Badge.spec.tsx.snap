// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should have hover/active styles only when the onClick handler is provided 1`] = `
.circuit-0 {
  border-radius: 999999px;
  color: #FFFFFF;
  display: inline-block;
  padding: 2px 8px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 700;
  text-align: center;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  background-color: #D8DDE1;
  color: #0F131A;
  border: 0;
  outline: 0;
  cursor: pointer;
  -webkit-transition: background-color 120ms ease-in-out;
  transition: background-color 120ms ease-in-out;
}

.circuit-0:hover,
.circuit-0:active {
  background-color: #9DA7B1;
}

.circuit-0:focus {
  outline: 0;
  box-shadow: 0 0 0 4px #AFD0FE;
}

.circuit-0:focus::-moz-focus-inner {
  border: 0;
}

<button
  class="circuit-0"
/>
`;

exports[`Badge should have the correct circle styles 1`] = `
.circuit-0 {
  border-radius: 999999px;
  color: #FFFFFF;
  display: inline-block;
  padding: 2px 8px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 700;
  text-align: center;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  background-color: #D8DDE1;
  color: #0F131A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 2px 4px;
  height: 24px;
  width: 24px;
}

<div
  class="circuit-0"
/>
`;

exports[`Badge should render with danger styles 1`] = `
.circuit-0 {
  border-radius: 999999px;
  color: #FFFFFF;
  display: inline-block;
  padding: 2px 8px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 700;
  text-align: center;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  background-color: #DB4D4D;
  color: #FFFFFF;
}

<div
  class="circuit-0"
/>
`;

exports[`Badge should render with default styles 1`] = `
.circuit-0 {
  border-radius: 999999px;
  color: #FFFFFF;
  display: inline-block;
  padding: 2px 8px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 700;
  text-align: center;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  background-color: #D8DDE1;
  color: #0F131A;
}

<div
  class="circuit-0"
/>
`;

exports[`Badge should render with primary styles 1`] = `
.circuit-0 {
  border-radius: 999999px;
  color: #FFFFFF;
  display: inline-block;
  padding: 2px 8px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 700;
  text-align: center;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  background-color: #3388FF;
  color: #FFFFFF;
}

<div
  class="circuit-0"
/>
`;

exports[`Badge should render with success styles 1`] = `
.circuit-0 {
  border-radius: 999999px;
  color: #FFFFFF;
  display: inline-block;
  padding: 2px 8px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 700;
  text-align: center;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  background-color: #47995A;
  color: #FFFFFF;
}

<div
  class="circuit-0"
/>
`;

exports[`Badge should render with warning styles 1`] = `
.circuit-0 {
  border-radius: 999999px;
  color: #FFFFFF;
  display: inline-block;
  padding: 2px 8px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 700;
  text-align: center;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  background-color: #F6CC1B;
  color: #0F131A;
}

<div
  class="circuit-0"
/>
`;
